project/views.py

```diff
@@ -1,3 +1,72 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from datetime           import datetime, timedelta
+
+from django.http        import JsonResponse
+from django.views       import View
+
+from my_settings        import SECRET_KEY, ALGORITHM
+from user.utils         import user_decorator
+from .models            import Category, Project, Gift, Community, Story, Like
+from user.models        import User
+from message.models     import Message
+
+class ProjectDetailView(View):
+    @user_decorator
+    def get(self, request, project_uri):
+        today = datetime.now()
+
+        if not Project.objects.filter(project_uri=project_uri).exists():
+            return JsonResponse({'message': 'PROJECT_NOT_EXIST'}, status=404)
+
+        project = Project.objects.get(project_uri=project_uri)
+
+        total_price = []
+        project_info = {
+            'category'         : project.category.name,
+            'name'             : project.name,
+            'thumbnail_url'    : project.thumbnail_url,
+            'creator'          : project.user.fullname,
+            'achieved_rate'    : project.achieved_rate,
+            'total_amount'     : project.total_amount,
+            'rest_date'        : (project.closing_date - today).days,
+            'total_supporters' : project.total_supporters,
+            'goal_amount'      : project.goal_amount,
+            'payment_date'     : project.closing_date + timedelta(days=1),
+            'like'             : False if (request.user==None or not project.like_set.filter(user_id=request.user.id).exists()) else True,
```

Comment:
```suggestion
            'like'             : False if (request.user is None or not project.like_set.filter(user_id=request.user.id).exists()) else True,
```
None을 판단 할 때는 is를 사용합니다.

Reply:
수정했습니다~~